elastic/kibana
# Delete 'usersEnabled' feature flag

issue: https://github.com/elastic/kibana/issues/133508

## Summary
It deletes `usersEnabled` feature flag and refactors the code.

* Delete all code for the case that the feature is disabled
* Move HostsKpiAuthentications to the user folder

### Impact
* Feature flag was deleted
* For the end-user everything should work as before


### How to test it
* Make sure that The User page, User details page and Hosts page works as before.
* Check if the user page is present on the global search and side navigation.
* The authentication tab should not be present on the Hots page.


### Checklist

Delete any items that are not applicable to this PR.
